lib/charliecloud.py

```diff
@@ -602,10 +752,10 @@ def path_full(self):
 
    @property
    def version(self):
-      if (self.tag is not None):
-         return self.tag
       if (self.digest is not None):
          return "sha256:" + self.digest
+      if (self.tag is not None):
+         return self.tag
```

Comment:
Why did this move?

Reply:
To prioritize digest over tag. For example:

```
$ ch-grow pull --arch=arm64/v8 centos:8
```

In the above our image ref will target `/v2/library/centos:8`. To get the `arm64` variant we need to  prioritize the hash given to us from the fat manifest.

Reply:
Aha, let's a comment to that effect then.